File: backend/src/main/java/com/cqupt/nepmserver/po/Supervisor.java
package com.cqupt.nepmserver.po;

import lombok.Data;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableId;

@Data
@TableName("supervisor")
public class Supervisor {
    @TableId(value = "tel_id")
    private String telId; //电话号码
    private String password; //
    private String realName; //真实姓名
    private String birthday; //出生日期
    private Integer sex;
    private String remarks; //备注信息
}

File: NEPS公众监督员端/src/api/supervisor.ts
import request from '@/utils/request'

export interface Supervisor {
  telId: string
  password: string
  realName?: string
  birthday?: string
  sex?: number
  remarks?: string
}

// 公众监督员登录
export const login = (data: { telId: string; password: string }) => {
  return request.post('/supervisor/getSupervisorByIdByPass', data)
}

// 公众监督员注册
export const register = (data: Supervisor) => {
  return request.post('/supervisor/saveSupervisor', data)
}

// 根据手机号查询公众监督员
export const getSupervisorById = (telId: string) => {
  return request.post('/supervisor/getSupervisorById', { telId })
}
